pkg/credentialprovider/azure/azure_acr_helper.go

```diff
@@ -75,7 +78,10 @@ const userAgent = "kubernetes-credentialprovider-acr"
 
 const dockerTokenLoginUsernameGUID = "00000000-0000-0000-0000-000000000000"
 
-var client = &http.Client{}
+var client = &http.Client{
+	Transport: utilnet.SetTransportDefaults(&http.Transport{}),
+	Timeout:   time.Second * 10,
```

Comment:
so current default value is never timeout? 10s may be too small compared to never timeout?